lib/bolt_server/transport_app.rb

```diff
@@ -630,6 +634,28 @@ def make_winrm_target(target_hash)
       [400, e.message]
     end
 
+    # Returns a list of targets parsed from a Project inventory
+    #
+    # @param project_ref [String] the project_ref to compute the inventory from
+    get '/project_inventory_targets' do
+      bolt_config = config_from_project(params['project_ref'])
+      if bolt_config.inventoryfile && bolt_config.project.inventory_file.to_s != bolt_config.inventoryfile
+        raise Bolt::ValidationError, "Project inventory must be defined in #{bolt_config.project.inventory_file}"
+      end
+      plugin = Bolt::Plugin.setup(bolt_config, nil)
+      inventory = Bolt::Inventory.from_config(bolt_config, plugin)
```

Comment:
Not sure I like that approach because it means we'll also need to remember to do https://github.com/puppetlabs/bolt/blob/main/lib/bolt/inventory.rb#L72-L75 when we get to Inventory plugins.

Re: "any plugins running now", I think that will still be an issue due to https://github.com/puppetlabs/bolt/blob/main/lib/bolt/inventory/group.rb#L21 but it is something we will eventually fix before releasing this so doesn't seem like a big deal.

I do agree that the `BOLT_INVENTORY` env. var bit makes `from_config` annoying to use. But I do like that when `BOLT_INVENTORY` is not set, `from_config` ensures I get an Inventory object constructed from my config that's ready for me to use (so I don't have to worry about e.g. resolving plugin references in the bolt config).

Since we won't be running into the `BOLT_INVENTORY` env. var issue in our current work, maybe we could keep `from_config` for now?

I do think it'd be worth adding a `Bolt::Inventory.from_config_strict(bolt_config, plugin)` method to Bolt that will load the inventory strictly from the config. That would satisfy our use-case and avoid the `BOLT_INVENTORY` environment variable issue.

Reply:
If there are other reasons to refactor this that seems fine.

But I don't think we need to worry about misconfiguration from user environment vars as a top concern. "I updated my replicated configuration so that the orchestrator container instance runs with the `BOLT_INVENTORY` env var set" isn't something I expect we'll find often. 

If a user _did_ start trying to pass env vars to the running container instances, the fix is to have them stop doing that.

Reply:
I don't think it's a huge deal. It's mostly reflective of a larger problem which is that there are cases where we can't currently avoid Bolt-specific logic that is potentially harmful or at least unneeded.

Reply:
That's fair, if our argument is for better code maintainability and less coupling to bolt specific logic I think that's a reasonable reason to look at refactors.

I just wanted to make sure our primary concerns don't center on problems we are unlikely to have.